libs/feature/record/src/lib/data-view-web-component/data-view-web-component.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { DataViewWebComponentComponent } from './data-view-web-component.component'
+import { BehaviorSubject, firstValueFrom } from 'rxjs'
+import { Configuration } from '@geonetwork-ui/data-access/gn4'
+import { MdViewFacade } from '../state'
+import { TranslateModule } from '@ngx-translate/core'
+import { Component, Input } from '@angular/core'
+
+const chartConfig1 = {
+  aggregation: 'sum',
+  xProperty: 'anneeappro',
+  yProperty: 'nbre_com',
+  chartType: 'bar',
+}
+
+const chartConfig2 = {
+  aggregation: 'min',
+  xProperty: 'pro',
+  yProperty: 'number',
+  chartType: 'line',
+}
+
+const metadata = {
+  uuid: 'md_record_1234',
+}
+
+class MdViewFacadeMock {
+  chartConfig$ = new BehaviorSubject(chartConfig1)
+  metadata$ = new BehaviorSubject(metadata)
+}
+class ConfigMock {
+  basePath: 'http://gn-api.url/'
+}
+
+@Component({
+  selector: 'gn-ui-copy-text-button',
+  template: '<div></div>',
+})
+export class MockCopyTextButtonComponent {
+  @Input() text: string
+  @Input() tooltipText: string
+  @Input() rows: number
+}
+
+describe('DataViewWebComponentComponent', () => {
+  let component: DataViewWebComponentComponent
+  let fixture: ComponentFixture<DataViewWebComponentComponent>
+  let facade
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        DataViewWebComponentComponent,
+        MockCopyTextButtonComponent,
+      ],
+      imports: [TranslateModule.forRoot()],
+      providers: [
+        {
+          provide: Configuration,
+          useClass: ConfigMock,
+        },
+        {
+          provide: MdViewFacade,
+          useClass: MdViewFacadeMock,
+        },
+      ],
+    }).compileComponents()
+    facade = TestBed.inject(MdViewFacade)
+    fixture = TestBed.createComponent(DataViewWebComponentComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+  })
+
+  describe('init webComponentHtml$', () => {
+    it('should generate HTML based on configs', async () => {
+      const html = await firstValueFrom(component.webComponentHtml$)
+      expect(html).toBe(
+        `<script src="https://cdn.jsdelivr.net/gh/geonetwork/geonetwork-ui@wc-dist/gn-wc.js"></script>
+        <gn-dataset-view-chart
+          api-url="http://localhost/undefined"
+          dataset-id="${metadata.uuid}"
+          aggregation="${chartConfig1.aggregation}"
+          x-property="${chartConfig1.xProperty}"
+          y-property="${chartConfig1.yProperty}"
+          chart-type="${chartConfig1.chartType}"
+          primary-color="#0f4395"
+          secondary-color="#8bc832"
+          main-color="#555"
+          background-color="#fdfbff"
+          main-font="'Inter', sans-serif"
+          title-font="'DM Serif Display', serif"
+        ></gn-dataset-view-chart>`
```

Comment:
The indentation here makes the end result a little bit confusing:
![image](https://github.com/geonetwork/geonetwork-ui/assets/10629150/5bd12e7e-ab96-4977-9ab4-5f3a8e3d848e)

Something like this would be more readable IMO:
```
<script src="https://cdn.jsdelivr.net/gh/geonetwork/geonetwork-ui@wc-dist/gn-wc.js"></script>
<gn-dataset-view-chart
    api-url="http://localhost:4200/geonetwork/srv/api"
    dataset-id="ded8164c-d56b-4ae0-b93d-459fc7c99a01"
    aggregation="sum"
    x-property="insee_com"
    y-property="superf_km"
    ...
></gn-dataset-view-chart>
```

Reply:
Good point!